test/library/standard/Reflection/getFieldRef-borrowed.chpl

```diff
@@ -0,0 +1,26 @@
+// this is a library function `Reflection.getFieldRef`, specifically.
+import Reflection;
+
+class FieldType {
+  var x: int;
+  proc setup() {
+    ref xRef = Reflection.getFieldRef(this, 0);
+    xRef = 20;
+  }
+}
+
+record MainType {
+  var field: owned FieldType?;
```

Comment:
does this still work if `FieldType` is `shared`? I'm pretty sure it will, but it might be good to have a variation of this test that checks that.

Reply:
It does. https://github.com/chapel-lang/chapel/pull/24829/commits/95dd68536f75942476fec45be7e7bcda6460098c expands the test.